onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -692,13 +696,27 @@ def create_multihead_attention_node(
         else:
             mha_inputs.append("")
 
+        if (
+            past_k in graph_input_names
+            and past_v in graph_input_names
+            and present_k in graph_output_names
+            and present_v in graph_output_names
+        ):
+            kv_cache_name_match = True
+        else:
+            kv_cache_name_match = False
+
         # Add optional inputs for MHA
-        if past_k and past_v and past_k in graph_input_names and past_v in graph_input_names:
+        if past_k and past_v and kv_cache_name_match:
+            mha_inputs.extend([key_padding_mask, add_qk, past_k, past_v])
```

Comment:
Although not introduced it this PR, the logic is questionable. 
why appending add_qk input depends on past_k and past_v is graph inputs?

Reply:
It should not depend on past_k and past_v values that should be added separately, also key_padding_mask should be separate right?

Reply:
I updated this but whisper tests are failing because of this. Will address this issue in another PR since it is unrelated to this one